cmd/scollector/collectors/ifstat_linux.go

```diff
@@ -106,3 +112,162 @@ func c_ifstat_linux() (opentsdb.MultiDataPoint, error) {
 	})
 	return md, err
 }
+
+const (
+	linuxNetBondSlaveIsUpDesc = "The status of the bonded or teamed interface."
+	linuxNetBondSlaveCount    = "The number of slaves on the bonded or teamed interface."
+)
+
+func c_if_bond_linux() (opentsdb.MultiDataPoint, error) {
+	var md opentsdb.MultiDataPoint
+	const bondingPath = "/proc/net/bonding"
+	bondDevices, _ := ioutil.ReadDir(bondingPath)
+	for _, fi := range bondDevices {
+		var iface string
+		var slaveCount int
+		if err := readLine(bondingPath+"/"+fi.Name(), func(s string) error {
+			f := strings.SplitN(s, ":", 2)
+			if len(f) != 2 {
+				return nil
+			}
+			f[0] = strings.TrimSpace(f[0])
```

Comment:
Agree on the checking error , but don't really feel like changing logic names etc on something that already works, this was just moved out of the proc_linux collector, but code is the same.